pkg/bootstrap/option/convert.go

```diff
@@ -138,7 +138,7 @@ func tlsContextConvert(tls *networkingAPI.ClientTLSSettings, sniName string, met
 		// No TLS.
 		return nil
 	}
-	if len(sniName) > 0 {
+	if len(tls.Sni) == 0 && tls.Mode == networkingAPI.ClientTLSSettings_ISTIO_MUTUAL {
```

Comment:
why do we not set SNI for mutual or simple?

Reply:
for ISTIO_MUTUAL we default it to "tracer" , "envoy_metrics_service" if `tls.Sni` is not specified - that was the existing behaviour. 
For Simple and Mutual we set it only if it is specified in TLS settings (because we do not know what to default to).

Are you suggesting we should not default from ISTIO_MUTUAL as well and only set if user sets it?